src/kernel/include/tcap.h

```diff
@@ -190,32 +204,28 @@ tcap_timer_update(struct cos_cpu_local_info *cos_info, struct tcap *next, tcap_t
 
 	/* next == INF? no timer required. */
 	left        = tcap_left(next);
-	if (timeout == TCAP_TIME_NIL && TCAP_RES_IS_INF(left)) return;
+	if (timeout == TCAP_TIME_NIL && TCAP_RES_IS_INF(left)) {
+		if (cos_info->timeout_next) {
+			chal_timer_disable();
+			cos_info->timeout_next = 0;
```

Comment:
Actually, I think it's okay to get rid of timeout_next variable, it's use is only here for now. I should have the logic inside chal_timer_disable() to not redo if it's already disabled. 

I thought of wraparound issues, but I was thinking it will take too long for a 64bit value to wraparound. But yes, I need to take care of it anyway.